### PR TITLE
Sleep tdd

### PR DESCRIPTION
**Pull request by demaceo**
### What does  this PR do?
Adds functionality to the `sleepRepository` class & creates tests for each newly created method.

### Where should the reviewer start?
View **sleepRepository.js** & **SleepRepository-test.js**

### How should this be manually tested?
Run `npm test` in Terminal.

### Any background context you want to provide?
Commented out code is still being utilized and will be removed upon next PR.

### CODE REVIEW

File: src/sleepRepository.js [hunk 1]
```diff
@@ -36,28 +33,43 @@ class SleepRepository {
   //   })
   // }
   // For a user, their sleep quality for a specific day (identified by a date)
-  findSleepQualityOnDate(date) {
-    this.records.find(record => {
-      if (record.date === date) {
-        return record.sleepQuality
-      }
-    })
+  findSleepQualityOn(date) {
```
Comment (peytonjo) [inline src/sleepRepository.js:36]: looks good!

File: src/sleepRepository.js [hunk 2]
```diff
@@ -36,28 +33,43 @@ class SleepRepository {
...
+      return this.weeklyRecords.push(`{Week ${weekCounter}: ${week}}`);
     })
+    // console.log(this.weeklyRecords);
   }
+
+  // determineWeeklySchedule() {
+  //   let mockRecords = this.records;
+  //   let week;
+  //   let weekCounter = 0;
+  //   mockRecords.reduce((acc, record) => {
```
Comment (peytonjo) [inline src/sleepRepository.js:62]: some of this commented out code will need to be deleted  to dry things up

File: src/sleepRepository.js [hunk 3]
```diff
@@ -104,57 +117,60 @@ class SleepRepository {
...
+
   // Find all users who average a sleep quality greater than 3 for a given week (7 days)
   //- you should be able to calculate this for any week, not just the latest week
-  filterSleepQualityGreaterThanThree() {
-    let qualitySleep = [];
-    this.records.filter(record => {
-      if(record.sleepQuality > 3){
-        return qualitySleep.push(record)
+  determineSleepQualityGreaterThanThree(todaysDate) {
+    let day = Number(todaysDate.charAt(8) + todaysDate.charAt(9));
```
Comment (peytonjo) [inline src/sleepRepository.js:125]: this looks really good!

File: src/sleepRepository.js [hunk 4]
```diff
@@ -104,57 +117,60 @@ class SleepRepository {
...
-  //       sum += sleepAct.quality;
-  //     }
-  //     return sum;
-  //   }, 0) / 7).toFixed(1);
-  // }
+  // find the user's night of sleep....EVER!
+  findWorstNightsSleep() {
+    let worstSleep = this.records.sort((a, b) => {
+      a.sleepQuality - b.sleepQuality
+    })
```
Comment (peytonjo) [inline src/sleepRepository.js:172]: glad we deleted some of that extra code

File: src/sleepRepository.js (hunk 3 above)
Comment (JWoltjen) [inline src/sleepRepository.js:125]: This is a really difficult function to tackle

**Review by peytonjo — APPROVED** [review]

**Review by JWoltjen — APPROVED** [review]
Good work Demaceo let's keep at it!

**Pull request merged**